templates/terraform/constants/spanner_database.go.erb

```diff
@@ -0,0 +1,33 @@
+<%- # the license inside this block applies to this file
+	# Copyright 2020 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+// customizeDiff func for additional checks on google_spanner_database properties:
+func resourceSpannerDBDdlCustomDiff(diff *schema.ResourceDiff, meta interface{}) error {
```

Comment:
It would be nice to have a unit test for this. You can take a look at https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_compute_instance_template.go#L622-L648 for how to structure this function so that it can be tested, and https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_compute_instance_template_test.go#L164 for the test of that function.

Using this would also require making some changes to the mock [interface](https://github.com/hashicorp/terraform-provider-google/blob/master/google/utils.go#L27) and [implementation](https://github.com/hashicorp/terraform-provider-google/blob/master/google/test_utils.go#L64-L84) to be able to set `ForceNew` on the ResourceDiff.

Let me know if that's too much for you and I can push the commits to make it happen, or if you want to go over it.

Reply:
I will work on it and push the change

Reply:
made the change and added the unit test, should the unit tests need to have all the cases ?